leshan-client-core/src/main/java/org/eclipse/leshan/client/observer/LwM2mClientObserver.java

```diff
@@ -69,4 +69,8 @@ void onDeregistrationFailure(ServerIdentity server, DeregisterRequest request, R
             String errorMessage, Exception cause);
 
     void onDeregistrationTimeout(ServerIdentity server, DeregisterRequest request);
+
+    // ============== Unexpected Error Handling =================
+
+    void onUnexpectedErrorOccurred(Throwable unexpectedError);
```

Comment:
maybe `onUnexpectedError` is enough ?

Reply:
Thank you for the advice, I've not had confidence in this honestly :)

Reply:
fixed at 16c6cb7